costs/costs_route.go

```diff
@@ -56,128 +52,75 @@ type esQueryParams struct {
-func parseCriteriaQueryParam(queryParam []string, parsedParams esQueryParams) (esQueryParams, error) {
-	if len(queryParam) != 1 {
-		return parsedParams, fmt.Errorf("expected queryParam len of 1 but got %d", len(queryParam))
-	}
-	criteriaList := strings.Split(queryParam[0], ",")
-	var resCriterions []string
-	for _, criterion := range criteriaList {
-		if simpleCriterionMap[criterion] {
-			resCriterions = append(resCriterions, criterion)
-		} else if len(criterion) >= 5 && criterion[:4] == "tag:" && strings.Count(criterion, ":") == 1 {
-			return parsedParams, fmt.Errorf("tags not yet implemented")
-		} else {
-			return parsedParams, fmt.Errorf("error parsing criterion : %s", criterion)
-		}
-	}
-	parsedParams.aggregationParams = resCriterions
-	return parsedParams, nil
-}
-
-// parseQueryParams parses the queryParams arguments.
-// It parses those arguments into a esQueryParams struct to be used by the
-// GetElasticSearchParams function that will create the ElasticSearch request.
-func parseQueryParams(queryParams url.Values) (esQueryParams, error) {
-	var parsedParams esQueryParams
-	var err error
-	for key, val := range queryParams {
-		if parserFunc := formFieldsToHandlerFunc[key]; parserFunc != nil {
-			parsedParams, err = parserFunc(val, parsedParams)
-			if err != nil {
-				return parsedParams, err
+func validateCriteriaParam(parsedParams esQueryParams) error {
+	for _, criterion := range parsedParams.aggregationParams {
+		if !simpleCriterionMap[criterion] {
+			if len(criterion) >= 5 && criterion[:4] == "tag:" && strings.Count(criterion, ":") == 1 {
+				return fmt.Errorf("tags not yet implemented")
 			}
-		} else {
-			return parsedParams, fmt.Errorf("could not parse param : %s=%s", key, val)
+			return fmt.Errorf("Error parsing criterion : %s", criterion)
 		}
 	}
-	if parsedParams.dateBegin.IsZero() {
-		return parsedParams, fmt.Errorf("'end' is not set")
-	} else if parsedParams.dateEnd.IsZero() {
-		return parsedParams, fmt.Errorf("'until' is not set")
-	} else if len(parsedParams.aggregationParams) < 1 {
-		return parsedParams, fmt.Errorf("'by' should at least have one criterion")
-	}
-	return parsedParams, nil
+	return nil
 }
 
-func parseFormAndQueryParams(request *http.Request) (esQueryParams, error) {
-	l := jsonlog.LoggerFromContextOrDefault(request.Context())
-	err := request.ParseForm()
-	if err != nil {
-		l.Info("Error parsing form data : "+err.Error(), nil)
-		return esQueryParams{}, err
-	}
-	parsedParams, err := parseQueryParams(request.Form)
-	if err != nil {
-		l.Info("Error parsing user submitted forms : "+err.Error(), nil)
-		return esQueryParams{}, err
+// validateAwsAccounts will validate awsAccounts passed to it.
+// It checks that they are numbers that are 12 character long
+func validateAwsAccounts(parsedParams esQueryParams) error {
+	for _, account := range parsedParams.accountList {
+		if _, err := strconv.ParseInt(account, 10, 0); err != nil || len(account) != 12 {
```

Comment:
Can you please point me to the documentation which says an AWS account ID is always 12 digits long ?

Reply:
It's written here : https://docs.aws.amazon.com/general/latest/gr/acct-identifiers.html

Reply:
Thanks!